django_components/component.py

```diff
@@ -160,14 +161,33 @@ def validate_fills_and_slots_(
                     f"'{template.name}' of component '{self._component_name}'."
                 )
             slots[slot_name] = slot
+
         # All fill nodes must correspond to a declared slot.
         unmatchable_fills = fills.keys() - slots.keys()
-        if unmatchable_fills:
-            msg = (
-                f"Component '{self._component_name}' passed fill(s) "
-                f"refering to undefined slot(s). Bad fills: {list(unmatchable_fills)}."
-            )
-            raise TemplateSyntaxError(msg)
+        if not unmatchable_fills:
+            return
+        # There are some fills that don't correspond to any slot. Raise an exception with helpful error message.
+        unfilled_slots: set[str] = {s.name for s in all_slots} - fills.keys()
```

Comment:
Quickly double-checking this PR now, I'm not sure this syntax is correct for older Python versions. Shouldn't it be `typing.Set[str]`? I'm surprised that the tests seem to be passing. Am I wrong?

Reply:
We're currently not running any typing checks at all, as such a small part of the library is type checked, so I'm guessing that part is just ignored. If the tests pass in 3.6, I'm happy.